[0.78.0] go panic after upgrading from v0.74.0 to v0.78.0

```
helmfile --log-level=debug diff
processing file "helmfile.yaml" in directory "."
first-pass rendering starting for "helmfile.yaml.part.0": inherited=&{default map[] map[]}, overrode=<nil>
first-pass uses: &{default map[] map[]}
replaced <no value>s to workaround https://github.com/golang/go/issues/24963 to address https://github.com/roboll/helmfile/issues/553:
  strings.Join({
        ... // 46 identical lines
        "  ",
        "",
-       "- name: image-pull-secret-<no value>",
-       "  namespace: <no value>",
+       "- name: image-pull-secret-",
+       "  namespace: ",
        "  chart: ./credentials/image-pull-secret",
        "",
-       "- name: postgresql-secret-credential-<no value>",
-       "  namespace: <no value>",
+       "- name: postgresql-secret-credential-",
+       "  namespace: ",
        "  chart: ./credentials/postgresql-secret-credential",
        "",
        ... // 4 identical lines
  }, "\n")

envvals_loader: loaded
OMITTED
envvals_loader: loaded
OMITTED
Decrypting secret /environments/XXX/secrets.yaml
exec: helm secrets dec/environments/XXX/secrets.yaml --tiller-namespace kube-system --kube-context kubernetes-admin@aws@kubernetes
exec: helm secrets dec /environments/XXX/secrets.yaml --tiller-namespace kube-system --kube-context kubernetes-admin@aws@kubernetes: Decrypting /environments/XXX/secrets.yaml

Decrypting /environments/XXX/secrets.yaml

panic: reflect.Value.MapIndex: value of type interface {} is not assignable to type string [recovered]
        panic: reflect.Value.MapIndex: value of type interface {} is not assignable to type string

goroutine 1 [running]:
github.com/urfave/cli.HandleAction.func1(0xc000544e78)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/urfave/cli@v0.0.0-20160620154522-6011f165dc28/app.go:474 +0x287
panic(0x1843440, 0xc000406330)
        /usr/local/Cellar/go/1.12.6/libexec/src/runtime/panic.go:522 +0x1b5
reflect.Value.assignTo(0x1887560, 0xc000406310, 0x94, 0x19b7b02, 0x16, 0x1843440, 0x0, 0x1887560, 0xc000406310, 0x94)
        /usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:2339 +0x437
reflect.Value.MapIndex(0x188b820, 0xc0004238f0, 0x15, 0x1887560, 0xc000406310, 0x94, 0x1887560, 0xc000406320, 0x94)
        /usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:1104 +0x8c
github.com/imdario/mergo.deepMerge(0x188b820, 0xc0004238f0, 0x15, 0x1889300, 0xc0004220c0, 0x15, 0xc0005431d8, 0x1, 0xc000408060, 0x0, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/imdario/mergo@v0.3.6/merge.go:90 +0x46f
github.com/imdario/mergo.deepMerge(0x188b820, 0xc000010180, 0x195, 0x188b820, 0xc0004d8018, 0x195, 0xc0005431d8, 0x0, 0xc000408060, 0x195, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/imdario/mergo@v0.3.6/merge.go:115 +0x97d
github.com/imdario/mergo.merge(0x1825ae0, 0xc000010180, 0x1825ae0, 0xc0004d8018, 0xc000543450, 0x1, 0x1, 0x0, 0x4)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/imdario/mergo@v0.3.6/merge.go:251 +0x299
github.com/imdario/mergo.Merge(...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/imdario/mergo@v0.3.6/merge.go:206
github.com/roboll/helmfile/pkg/state.(*HelmState).loadEnvValues(0xc00000c780, 0x19ab17d, 0x7, 0xc0000d9bc0, 0x19df080, 0x19df510, 0xc000422c60, 0x1, 0xc000456000)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/state/create.go:225 +0xba5
github.com/roboll/helmfile/pkg/state.(*StateCreator).LoadEnvValues(0xc000424280, 0xc00000c1e0, 0x19ab17d, 0x7, 0xc0000d9bc0, 0xc00000c1e0, 0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/state/create.go:113 +0xc2
github.com/roboll/helmfile/pkg/state.(*StateCreator).ParseAndLoad(0xc000424280, 0xc0003bac00, 0x395, 0x400, 0x19a77b4, 0x1, 0xc0000dfa20, 0x14, 0x19ab17d, 0x7, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/state/create.go:146 +0x138
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).renderPrestate(0xc0000ab5c0, 0xc0000d9bc0, 0x19a77b4, 0x1, 0xc0000dfa20, 0x14, 0xc0003ca000, 0x1c03, 0x1e03, 0x0, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/two_pass_renderer.go:52 +0x454
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).twoPassRenderTemplateToYaml(0xc0000ab5c0, 0xc0000d9b00, 0x0, 0x19a77b4, 0x1, 0xc0000dfa20, 0x14, 0xc0003ca000, 0x1c03, 0x1e03, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/two_pass_renderer.go:96 +0xfb
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).renderTemplatesToYamlWithEnv(...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/two_pass_renderer.go:78
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).renderTemplatesToYaml(0xc0000ab5c0, 0x19a77b4, 0x1, 0xc0000dfa20, 0x14, 0xc0003ca000, 0x1c03, 0x1e03, 0xc0000d9ae0, 0x1, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/two_pass_renderer.go:74 +0xd2
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).renderAndLoad(0xc0000ab5c0, 0x0, 0x0, 0x19a77b4, 0x1, 0xc000348650, 0xd, 0xc0003ca000, 0x1c03, 0x1e03, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/desired_state_file_loader.go:165 +0x6d3
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).loadFileWithOverrides(0xc0000ab5c0, 0x0, 0x0, 0x19a77b4, 0x1, 0x19b0001, 0xd, 0xc0000e1a01, 0x199, 0x0, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/desired_state_file_loader.go:105 +0x1f4
github.com/roboll/helmfile/pkg/app.(*desiredStateLoader).Load(0xc0000ab5c0, 0x19b0001, 0xd, 0x21b5c88, 0x0, 0x0, 0x21b5c88, 0x0, 0x0, 0x19b0001, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/desired_state_file_loader.go:53 +0x48b
github.com/roboll/helmfile/pkg/app.(*App).loadDesiredStateFromYaml(0xc0003be000, 0x19b0001, 0xd, 0xc000274480, 0x1, 0x1, 0x0, 0x0, 0x21b5c88)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:244 +0x1d2
github.com/roboll/helmfile/pkg/app.(*App).visitStates.func1(0x19b0001, 0xd, 0xc0001555c0, 0x3c, 0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:257 +0x1be
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles.func1(0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:215 +0x9d
github.com/roboll/helmfile/pkg/app.(*App).within(0xc0003be000, 0x19a77b4, 0x1, 0xc0000e1980, 0xc0002747f8, 0x2)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:157 +0x725
github.com/roboll/helmfile/pkg/app.(*App).visitStateFiles(0xc0003be000, 0x0, 0x0, 0xc0000ab440, 0x20, 0x18fa0e0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:209 +0x29f
github.com/roboll/helmfile/pkg/app.(*App).visitStates(0xc0003be000, 0x0, 0x0, 0x21b5c88, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:250 +0xd4
github.com/roboll/helmfile/pkg/app.(*App).VisitDesiredStatesWithReleasesFiltered(0xc0003be000, 0x0, 0x0, 0xc00007e6b0, 0xc000274a01, 0xc00000eca0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:398 +0x426
github.com/roboll/helmfile/pkg/app.(*App).ForEachState(0xc0003be000, 0xc00000eca0, 0xc00007e6a0, 0x198b700)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:344 +0x81
github.com/roboll/helmfile/pkg/app.(*App).Diff(0xc0003be000, 0x1b552e0, 0xc00007e6a0, 0xc00007e6a0, 0x18d4880)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/pkg/app/app.go:102 +0x6e
main.main.func4(0xc0003be000, 0xc0003b4140, 0x0, 0xc00007e680, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/main.go:187 +0x6d
main.action.func1(0xc0003b4140, 0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/main.go:540 +0x121
reflect.Value.call(0x185b4a0, 0xc00007e1d0, 0x13, 0x19a8193, 0x4, 0xc000274e18, 0x1, 0x1, 0xc000090040, 0x1013d63, ...)
        /usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:447 +0x461
reflect.Value.Call(0x185b4a0, 0xc00007e1d0, 0x13, 0xc000274e18, 0x1, 0x1, 0xc0000bc750, 0xc0000bc7b8, 0x140)
        /usr/local/Cellar/go/1.12.6/libexec/src/reflect/value.go:308 +0xa4
github.com/urfave/cli.HandleAction(0x185b4a0, 0xc00007e1d0, 0xc0003b4140, 0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/urfave/cli@v0.0.0-20160620154522-6011f165dc28/app.go:483 +0x1ff
github.com/urfave/cli.Command.Run(0x19a84cb, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x19cf193, 0x35, 0x0, ...)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/urfave/cli@v0.0.0-20160620154522-6011f165dc28/command.go:186 +0x8d1
github.com/urfave/cli.(*App).Run(0xc000001500, 0xc0000d4000, 0x3, 0x3, 0x0, 0x0)
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/pkg/mod/github.com/urfave/cli@v0.0.0-20160620154522-6011f165dc28/app.go:237 +0x601
main.main()
        /private/tmp/helmfile-20190616-81645-qlrva7/helmfile-0.78.0/src/github.com/roboll/helmfile/main.go:393 +0x279c

```

--- a/pkg/state/create.go
+++ b/pkg/state/create.go
@@ -7,6 +7,7 @@ import (
 	"github.com/imdario/mergo"
 	"github.com/roboll/helmfile/pkg/environment"
 	"github.com/roboll/helmfile/pkg/helmexec"
+	"github.com/roboll/helmfile/pkg/maputil"
 	"go.uber.org/zap"
 	"gopkg.in/yaml.v2"
 	"io"
@@ -222,7 +223,14 @@ func (st *HelmState) loadEnvValues(name string, ctxEnv *environment.Environment,
 				if err := yaml.Unmarshal(bytes, &m); err != nil {
 					return nil, fmt.Errorf("failed to load environment secrets file \"%s\": %v", path, err)
 				}
-				if err := mergo.Merge(&envVals, &m, mergo.WithOverride); err != nil {
+				// All the nested map key should be string. Otherwise we get strange errors due to that
+				// mergo or reflect is unable to merge map[interface{}]interface{} with map[string]interface{} or vice versa.
+				// See https://github.com/roboll/helmfile/issues/677
+				vals, err := maputil.CastKeysToStrings(m)
+				if err != nil {
+					return nil, err
+				}
+				if err := mergo.Merge(&envVals, &vals, mergo.WithOverride); err != nil {
 					return nil, fmt.Errorf("failed to load \"%s\": %v", path, err)
 				}
 			}

